packages/workspace/src/tslint/nxEnforceModuleBoundariesRule.ts

```diff
@@ -134,16 +134,18 @@ class EnforceModuleBoundariesWalker extends Lint.RuleWalker {
       this.projectPath
     );
     const sourceProject = findSourceProject(this.projectGraph, filePath);
+    const targetImportProject = getTargetProjectBasedOnRelativeImport(
```

Comment:
If `targetImportProject` is defined, we can skip the `findProjectUsingImport` on line 159.

It should be something like:
```typescript
const targetProject = targetImportProject || findProjectUsingImport(
  this.projectGraph,
  this.targetProjectLocator,
  filePath,
  imp,
  this.npmScope
);
```